.github/workflows/build.yml

```
@@ -224,12 +224,12 @@ jobs:
             ${{ hashFiles('setup.py') }}"
           restore-keys: |
             ${{ env.BASE_CACHE_KEY }}
-      - name: Install dependencies
+      - name: Install build dependencies
         run: |
-          python -m pip install --upgrade pip wheel
-          pip install --upgrade --requirement requirements.txt
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade build
```

Is there a reason we can't/shouldn't consolidate this into a single line?
```suggestion
          python -m pip install --upgrade build pip setuptools wheel
```

---

At minimum we need to make sure `pip` is up to date before proceeding and we generally also upgrade `setuptools` and `wheel` as "core" packages at the same time. Installing `wheel` also ensures that for any later package installation `pip` will be able to build a wheel if one is not available for download, but this only works if it's already installed.

---

Thanks - can you please add a comment to that effect?

---

Where would you like a comment added? We follow that general process for `pip` package installation as seen in other places in this project:
https://github.com/cisagov/skeleton-python-library/blob/20920997de8b9c82e89013ea96abab690a56633d/setup-env#L146-L155
https://github.com/cisagov/skeleton-python-library/blob/20920997de8b9c82e89013ea96abab690a56633d/.github/workflows/build.yml#L98-L101
The previous build configuration:
https://github.com/cisagov/skeleton-python-library/blob/20920997de8b9c82e89013ea96abab690a56633d/.github/workflows/build.yml#L227-L230

---

I'd say to put the comment right here since this is where we are doing back-to-back `python -m pip install --upgrade` lines and that is where someone (likely future me) will wonder why we aren't consolidating them: https://github.com/cisagov/skeleton-python-library/blob/bc1adda0ef49fa49be0815f9c5f309775eaaff24/.github/workflows/build.yml#L228-L230

---

I guess my point is more that this is a recurring pattern in how we install Python packages. It's handled the same way in our Docker projects that use Python as an example. I'm averse to documenting such a global approach in one specific spot for one specific type of project.

---

I understand what you are saying, but do you have a better place to document this?

---

What about documenting this in the development guide?  That is the most global place we have for this sort of thang.

---

Development guide was my best alternative as well.

---

That sounds like an excellent place for this. 👍

---

@dav3r Is https://github.com/cisagov/development-guide/issues/75 suitable to proceed with this pull request for now?

---

Yes, thanks for creating that issue!